main.go

```diff
@@ -30,6 +30,12 @@ func init() {
 }
 
 func main() {
+	if err := realMain(); err != nil {
+		os.Exit(1)
+	}
```

Comment:
exitCodeみたいな型を定義する方法の方がいいがする

```suggestion
  type exitCode int
  
  const (
    statusOK exitCode = 0
    statusError       = 1
  )
  os.Exit(int(realMain()))
```

Reply:
なるべくGoのerrorで扱いたいからこんな風に扱うのはどうかな
```suggestion
type ExitError struct {
	err error
	exitCode int
}

cosnt (
	statusOK = iota
	statusError
)

func realMain() *ExitError {
```

Reply:
Goのエラー型で扱いたいならexitCodeのフィールドはいらないかな
```go
type ExitError truct {
   err error
}
```
こんな感じで，エラー起きたときはerrorを返すっていういつもの感じで実装してもらえれば．
ただ，エラーが発生したことを，main関数に伝える必要が今の実装だと必要ないからその場合エラー発生時のloggingをmain関数に移すとかしたほうがいいかもしれない